code/ship/ship.cpp

```diff
@@ -10334,6 +10334,8 @@ int ship_fire_primary(object * obj, int stream_weapons, int force)
 				if (winfo_p->wi_flags3 & WIF3_APPLY_RECOIL){
 					firepoint_list = new vec3d[numtimes * points];
 					vm_vec_zero(&total_impulse);
+				} else {
+					firepoint_list = nullptr;
```

Comment:
To me that looks like it's based on clang 3.4. I guess the workaround is simple, revert to using the NULL macro, but it's a disappointing result.